nova/scheduler/filters/resize_reservation_filter.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2020 SAP SE
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from oslo_log import log as logging
+
+import nova.conf
+from nova.scheduler import filters
+from nova.scheduler import utils
+
+LOG = logging.getLogger(__name__)
+
+CONF = nova.conf.CONF
+
+
+class ResizeReservedRAMFilter(filters.BaseHostFilter):
+    """Only return hosts for resize with sufficient available RAM."""
+
+    RUN_ON_REBUILD = False
+
+    def reserved_percent_ram(self):
+        return CONF.filter_scheduler.resized_reserved_ram_percent
+
+    def host_passes(self, host_state, request_spec):
+        if utils.request_is_resize(request_spec):
+            return True
+
+        reserved_ram_ratio = self.reserved_percent_ram() / 100.0
+        reserved_ram = host_state.total_usable_ram_mb * reserved_ram_ratio
+
+        # Apply reserved memory to total and free memory
+        host_state.total_usable_ram_mb -= reserved_ram
+        free_ram_mb = host_state.free_ram_mb - reserved_ram
+
+        memory_mb_limit = \
+            host_state.total_usable_ram_mb * host_state.ram_allocation_ratio
```

Comment:
We first reserve the RAM and then we apply the ram_allocation_ratio? We should document this, so a person just trying to configure this filter doesn't have to understand the code. Also document this in the option.

Reply:
Will do.